x-pack/platform/plugins/shared/alerting/server/application/gap_auto_fill_scheduler/methods/create/create_gap_auto_fill_scheduler.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Boom from '@hapi/boom';
+import type { SavedObjectsCreateOptions } from '@kbn/core/server';
+import type { RulesClientContext } from '../../../../rules_client/types';
+import type { CreateGapAutoFillSchedulerParams } from './types';
+import type { GapAutoFillSchedulerResponse } from '../../result/types';
+import { createGapAutoFillSchedulerSchema } from './schemas';
+import { transformGapAutoFillSchedulerCreateParamToSavedObject } from './transforms/transform_gap_auto_fill_scheduler_param_to_saved_object';
+import { transformSavedObjectToGapAutoFillSchedulerResult } from '../../transforms';
+import { GAP_AUTO_FILL_SCHEDULER_SAVED_OBJECT_TYPE } from '../../../../saved_objects';
+import type { GapAutoFillSchedulerSO } from '../../../../data/gap_auto_fill_scheduler/types/gap_auto_fill_scheduler';
+import { WriteOperations, AlertingAuthorizationEntity } from '../../../../authorization';
+import {
+  gapAutoFillSchedulerAuditEvent,
+  GapAutoFillSchedulerAuditAction,
+} from '../../../../rules_client/common/audit_events';
+import { GAP_AUTO_FILL_SCHEDULER_TASK_TYPE } from '../../../../lib/rule_gaps/types/scheduler';
+
+export async function createGapAutoFillScheduler(
+  context: RulesClientContext,
+  params: CreateGapAutoFillSchedulerParams
+): Promise<GapAutoFillSchedulerResponse> {
+  try {
+    createGapAutoFillSchedulerSchema.validate(params);
+  } catch (error) {
+    const { request: _omitRequest, ...otherParams } = params;
+    throw Boom.badRequest(
+      `Error validating gap auto fill scheduler parameters - ${(error as Error).message}
+Payload summary: ${JSON.stringify(otherParams, (key, value) =>
+        key === 'request' ? undefined : value
+      )}`
+    );
+  }
+
+  try {
+    for (const ruleType of params.ruleTypes) {
+      await context.authorization.ensureAuthorized({
+        ruleTypeId: ruleType.type,
+        consumer: ruleType.consumer,
+        operation: WriteOperations.CreateGapAutoFillScheduler,
+        entity: AlertingAuthorizationEntity.Rule,
+      });
+    }
+  } catch (error) {
+    context.auditLogger?.log(
+      gapAutoFillSchedulerAuditEvent({
+        action: GapAutoFillSchedulerAuditAction.CREATE,
+        error,
+      })
+    );
+    throw error;
+  }
+
+  try {
+    const soClient = context.unsecuredSavedObjectsClient;
+    const taskManager = context.taskManager;
+
+    // Throw error if a gap auto fill scheduler already exists for the same (rule type, consumer) pair
+    const pairs = Array.from(new Set(params.ruleTypes.map((rt) => `${rt.type}:${rt.consumer}`)));
```

Comment:
there's no unit test for this error path...I haven't looked at the functional tests yet but if there's no test for that there, we should add a test somewhere.